File: AlgorithmCoderbyte/LeetCode C-sharp/_0013RomanToInteger.cs
﻿using System;

namespace AlgorithmCoderbyte.LeetCode_C_sharp
{
    class _0013RomanToInteger
    {
        public static int RomanToInt(string s)
        {

            var current = GetNumber(s[0]);
            var total = 0;
            for (int i = 1; i < s.Length; i++)
            {
                var next = GetNumber(s[i]);
                if (current < next)
                {
                    total -= current;
                }
                else
                {
                    total += current;
                }
                current = next;
            }
            total += current;
            return total;
        }

        public static int GetNumber(char ch)
        {
            switch (ch)
            {
                case 'I': return 1;
                case 'V': return 5;
                case 'X': return 10;
                case 'L': return 50;
                case 'C': return 100;
                case 'D': return 500;
                case 'M': return 1000;
                default: return 0;
            }
        }

        public static void Run()
        {
            Console.WriteLine(_0013RomanToInteger.RomanToInt("IV"));
        }
    }
}

File: AlgorithmCoderbyte/LeetCode TypeScript/0013 Roman to Integer.ts
function romanToInt(s: string): number {
    let total: number = 0;

    let cur: number = getNumber(s.charAt(0));

    for (let i = 1; i < s.length; i++) {
        let next = getNumber(s.charAt(i));
        if (cur < next) {
            total -= cur;
        }
        else {
            total += cur;
        }
        cur = next;
    }

    total += cur;


    return total;



    function getNumber(s: string): number {
        switch (s) {
            case "I": return 1;
            case "V": return 5;
            case "X": return 10;
            case "L": return 50;
            case "C": return 100;
            case "D": return 500;
            case "M": return 1000;
            default: return 0;
        }
    }

};
console.log(romanToInt("IV"));